diff -U3 -Narp binutils-2.37.orig/libiberty/rust-demangle.c binutils-2.37/libiberty/rust-demangle.c
--- binutils-2.37.orig/libiberty/rust-demangle.c	2021-07-15 16:19:46.000000000 +0200
+++ binutils-2.37/libiberty/rust-demangle.c	2021-07-19 09:50:29.003790488 +0200
@@ -75,10 +75,10 @@ struct rust_demangler
   int version;
 
   /* Recursion depth.  */
-  uint recursion;
+  unsigned int recursion;
   /* Maximum number of times demangle_path may be called recursively.  */
 #define RUST_MAX_RECURSION_COUNT  1024
-#define RUST_NO_RECURSION_LIMIT   ((uint) -1)
+#define RUST_NO_RECURSION_LIMIT   ((unsigned int) -1)
 
   uint64_t bound_lifetime_depth;
 };
